test/e2e/mob_test.go

```diff
@@ -0,0 +1,54 @@
+package e2e_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/openshift/machine-config-operator/test/framework"
+	"github.com/openshift/machine-config-operator/test/helpers"
+	"github.com/stretchr/testify/require"
+
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+)
+
+func TestMachineOSBuilder(t *testing.T) {
+	cs := framework.NewClientSet("")
+	mcpName := "test-mcp"
+	namespace := "openshift-machine-config-operator"
+	mobPodNamePrefix := "machine-os-builder"
+
+	helpers.CreateMCP(t, cs, mcpName)
+	time.Sleep(5 * time.Second) // Wait a bit to ensure MCP is fully created
+
+	// added retry because another process was modifying the MCP concurrently
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+
+		// Get the latest MCP
+		mcp, err := cs.MachineConfigPools().Get(context.TODO(), mcpName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		// Set the label
+		mcp.ObjectMeta.Labels[ctrlcommon.LayeringEnabledPoolLabel] = ""
+
+		// Try to update the MCP
+		_, err = cs.MachineConfigPools().Update(context.TODO(), mcp, metav1.UpdateOptions{})
+		return err
+	})
+	require.Nil(t, retryErr)
+
+	// wait for Machine OS Builder pod to start
+	helpers.WaitForPodStart(t, cs, mobPodNamePrefix, namespace)
+
+	// delete the MachineConfigPool
+	cs.MachineConfigPools().Delete(context.TODO(), mcpName, metav1.DeleteOptions{})
+
+	time.Sleep(5 * time.Second)
+
+	// wait for Machine OS Builder pod to stop
+	helpers.WaitForPodStop(t, cs, mobPodNamePrefix, namespace)
```

Comment:
I don't know that it matters if we leak a pod here, but if this fails it would imply that our "off" switch doesn't work, so I assume we would want to check the return here too?